File: ios/Kwift/com/lightningkite/kwift/views/shared/EntryPoint.swift
//Package: com.lightningkite.kwift.views.shared
//Converted using Kwift2

import Foundation



public protocol EntryPoint {
    
    func handleDeepLink(schema: String, host: String, path: String, params: Dictionary<String, String>) -> Void
    func handleDeepLink(_ schema: String, _ host: String, _ path: String, _ params: Dictionary<String, String>) -> Void
    
    func onBackPressed() -> Bool
    
    var mainStack: ObservableStack<ViewGenerator>?  { get }
}

public extension EntryPoint {
    
    func handleDeepLink(schema: String, host: String, path: String, params: Dictionary<String, String>) -> Void {
        print("Empty handler")
    }
    func handleDeepLink(_ schema: String, _ host: String, _ path: String, _ params: Dictionary<String, String>) -> Void {
        return handleDeepLink(schema: schema, host: host, path: path, params: params)
    }
    
    func onBackPressed() -> Bool {
        return false
    }
    public var mainStack: ObservableStack<ViewGenerator>?  {
        get {
            return nil
        }
    }
}
 

File: android/src/main/java/com/lightningkite/kwift/views/shared/EntryPoint.kt
package com.lightningkite.kwift.views.shared

import com.lightningkite.kwift.observables.shared.ObservableStack

interface EntryPoint {
    fun handleDeepLink(schema: String, host: String, path: String, params: Map<String, String>){
        println("Empty handler")
    }
    fun onBackPressed(): Boolean = false
    val mainStack: ObservableStack<ViewGenerator>? get() = null
}
